magda-web-client/src/UI/DataPreviewMap.js

```diff
@@ -1,18 +1,88 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import './DataPreviewMap.css';
+import { config } from '../config'
+
+const defaultDataSourcePreference = ['GeoJSON','WFS','WMS','csv-geo-au','KML'];
+
+function determineDistribution(props)
+{
+    const distributions = props.dataset.distributions;
+    let dataSourcePreference = props.dataset.dataSourcePreference;
+    if(!dataSourcePreference || !dataSourcePreference.length) dataSourcePreference = defaultDataSourcePreference;
+    dataSourcePreference = dataSourcePreference.map(item => item.toLowerCase());
+    if(!distributions || !distributions.length) return null;
+    let selectedDis = null, perferenceOrder = -1;
+    distributions.forEach(dis => {
+        const format = dis.format.toLowerCase();
+        const disPerORder = dataSourcePreference.indexOf(format);
+        if(disPerORder == -1) return;
+        if(perferenceOrder == -1 || disPerORder<perferenceOrder) {
+            perferenceOrder = disPerORder;
+            selectedDis = dis;
+            return;
+        }
+    });
+    return selectedDis;
+}
+
+function createCatalogItemFromDistribution(dis)
+{
+    return {
+        "version": "0.0.05",
+        "initSources": [
+            {
+                "catalog": [
+                    {
+                        "name": dis.title,
+                        "type": "magda-item",
+                        "url": "/",
+                        "distributionId": dis.identifier,
+                        "isEnabled": true,
+                        "zoomOnEnable": true
+                    }
+                ],
+                "baseMapName": "Positron (Light)"
+            }
+        ]
+    };
+}
 
 function DataPreviewMap(props) {
-  // Go crazy jacky!
-  return (
-    <div className="data-preview-map">
-      <h3>Map Preview</h3>
-    </div>
-  );
+    
+    const { identifier } = props.dataset;
+    if(identifier=='') return (
+        <div className="data-preview-map">
+            <h3>Map Preview</h3>
+            Loading....
+        </div>
+    );
+
+    const selectedDistribution = determineDistribution(props);
+
+    if(!selectedDistribution) return (
+        <div className="data-preview-map">
+            No map preview available
+        </div>
+    );
+
+    const catalogItem = createCatalogItemFromDistribution(selectedDistribution);
+    const previewMapUrl = config.previewMapUrl + "#start=" + encodeURIComponent(JSON.stringify(catalogItem));
+    return (
+        <div className="data-preview-map">
+            <h3>Map Preview</h3>
+            <div className='mui-row'>
```

Comment:
Why do we need the `mui-row` and `mui-col-sm-12` div here?

Reply:
@chloeleichen Saw somewhere in your chart component --- copied here and thought it might help to keep looking consistent with yours. Do you think it's better to put it as:
```
<div className="data-preview-map">
                <h3>Map Preview</h3>
                    <iframe 
                        title={this.state.selectedDistribution.title}
                        width="100%"
                        height="420px"
                        frameBorder="0"
                        src={(config.previewMapUrl + "#mode=preview&hideExplorerPanel=1")}
                        ref={f=>this.iframeRef=f}
                    />
            </div>
```